sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,414 @@
+      "\n";
+
+    const fixtures = nock.recorder.play();
+
+    // Create the directories recursively incase they don't exist
+    try {
+      // Stripping away the filename from the filepath and retaining the directory structure
+      fs.ensureDirSync(
+        "./recordings/" + this.filepath.substring(0, this.filepath.lastIndexOf("/") + 1)
+      );
+    } catch (err) {
+      if (err.code !== "EEXIST") throw err;
+    }
+
+    const file = fs.createWriteStream("./recordings/" + this.filepath, {
+      flags: "w"
+    });
+
+    // Some tests expect errors to happen and, if a writing error is thrown in one of these tests, it may be captured in a catch block by accident,
+    // resulting in unexpected behavior. For this reason we're printing it to the console as well
+    file.on("error", (err: any) => {
+      console.log(err);
+      throw err;
+    });
+
+    file.write(importNockStatement);
+
+    // Saving the recording to the file
+    for (const fixture of fixtures) {
+      // We're not matching query string parameters because they may contain sensitive information, and Nock does not allow us to customize it easily
+      const updatedFixture = fixture.toString().replace(/\.query\(.*\)/, ".query(true)");
+      file.write(this.filterSecrets(updatedFixture) + "\n");
+    }
+
+    file.end();
+
+    nock.recorder.clear();
+    nock.restore();
+  }
+}
+
+// To better understand how this class works, it's necessary to comprehend how HTTP async requests are made:
+// A new request object is created
+//    let req = new XMLHttpRequest();
+// The request is opened with some important information
+//    req.open(method, url, async, user, password);
+// Since we're dealing with an async request, we must set a way to know when the response is ready
+//    req.onreadystatechange = function() {
+//      if (req.readyState === 4) do_something;
+//    }
+// Finally, the request is sent to the server
+//    req.send(data);
+
+// Nise module does not have a native implementation of record/playback like Nock does
+// This class overrides requests' 'open', 'send' and 'onreadystatechange' functions, adding our own code to them to deal with requests
+export class NiseRecorder extends BaseRecorder {
+  private readonly sasQueryParameters = ["se", "sig", "sp", "spr", "srt", "ss", "st", "sv"];
+  private recordings: any[] = [];
+
+  constructor(testSuiteTitle: string, testTitle: string) {
+    super("browsers", testSuiteTitle, testTitle);
+  }
+
+  // Inserts a request/response pair into the recordings array
+  private async recordRequest(request: any, data: any): Promise<void> {
+    const responseHeaders: any = {};
+    const responseHeadersPairs = request.getAllResponseHeaders().split("\r\n");
+    for (const pair of responseHeadersPairs) {
+      const [key, value] = pair.split(": ");
+      responseHeaders[key] = value;
+    }
+
+    // We're not storing SAS Query Parameters because they may contain sensitive information
```

Comment:
It seems like the service should send you the additional set of parameters that may contain sensitive information

Reply:
Updated